cli/src/commands/get.rs

```diff
@@ -176,7 +176,16 @@ pub fn run(get_args: &GetArgs, client: Client, printer: &Printer) -> Result<()>
                 });
                 sources
             };
-            printer.print_resources(&sources)?;
+            let mut enriched_sources: Vec<EnrichedSource> = Vec::with_capacity(sources.len());
+            for source in sources.into_iter() {
+                let bucket = match source.bucket_id.as_ref() {
+                    // User may not have permission to fetch bucket, or bucket may not exist
+                    Some(bucket_id) => client.get_bucket(bucket_id.clone()).ok(),
+                    None => None,
+                };
+                enriched_sources.push(EnrichedSource { source, bucket });
```

Comment:
As a rule, constructing collections via mutation is an anti-pattern in Rust, and rarely necessary. The appearance of a `let mut` should always arouse suspicion. This can be rewritten without mutation as, say:
```rust
let enriched_sources: Vec<EnrichedSource> = sources
    .into_iter()
    .map(|source| {
        let bucket = source
            .bucket_id
            .as_ref()
            .and_then(|id| client.get_bucket(id.clone()).ok());
        EnrichedSource { source, bucket }
    })
    .collect();
```